Makefile cannot find brotli-related dynamic libraries MacOS

The makefile currently terminates with error 2 when trying to compile USDX step by step following the instructions in the readme. 

### Actual behaviour

First the makefile runs through the creation of UltraStarDeluxe.app without issues. Then it gets to the point where the output ends with:

``` Mac OS X application created (UltraStarDeluxe.app).
Using brew with libdir /usr/local/lib
Please report issues to the developer team, preferably mischi.
Have fun.


Creating the standalone Mac OS X application
```
here it starts throwing errors

```
error: /Library/Developer/CommandLineTools/usr/bin/otool-classic: can't open file: @loader_path/libbrotlicommon.1.dylib (No such file or directory)
error: /Library/Developer/CommandLineTools/usr/bin/otool-classic: can't open file: @loader_path/libbrotlicommon.1.dylib (No such file or directory)
work on ./game/ultrastardx
change the install name for /opt/homebrew/opt/sdl2/lib/libSDL2-2.0.0.dylib
change the install name for /opt/homebrew/opt/sdl2_image/lib/libSDL2_image-2.0.0.dylib
change the install name for /opt/X11/lib/libfreetype.6.dylib
change the install name for /opt/homebrew/opt/portaudio/lib/libportaudio.2.dylib
change the install name for /opt/homebrew/opt/ffmpeg@4/lib/libavcodec.58.dylib
change the install name for /opt/homebrew/opt/ffmpeg@4/lib/libavformat.58.dylib
change the install name for /opt/homebrew/opt/ffmpeg@4/lib/libavutil.56.dylib
change the install name for /opt/homebrew/opt/ffmpeg@4/lib/libswresample.3.dylib
change the install name for /opt/homebrew/opt/ffmpeg@4/lib/libswscale.5.dylib
change the install name for /opt/homebrew/opt/lua/lib/liblua.dylib
work on /opt/homebrew/opt/brotli/lib/libbrotlidec.1.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: UltraStarDeluxe.app/Contents/MacOS/libbrotlidec.1.dylib
change the install name for /opt/homebrew/opt/brotli/lib/libbrotlidec.1.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: UltraStarDeluxe.app/Contents/MacOS/libbrotlidec.1.dylib
change the install name for @loader_path/libbrotlicommon.1.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: UltraStarDeluxe.app/Contents/MacOS/libbrotlidec.1.dylib
work on /opt/homebrew/opt/brotli/lib/libbrotlienc.1.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: UltraStarDeluxe.app/Contents/MacOS/libbrotlienc.1.dylib
change the install name for /opt/homebrew/opt/brotli/lib/libbrotlienc.1.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: UltraStarDeluxe.app/Contents/MacOS/libbrotlienc.1.dylib
change the install name for @loader_path/libbrotlicommon.1.dylib
/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: UltraStarDeluxe.app/Contents/MacOS/libbrotlienc.1.dylib
make[1]: *** No rule to make target `@loader_path/libbrotlicommon.1.dylib', needed by `UltraStarDeluxe.app/Contents/MacOS/libbrotlicommon.1.dylib'.  Stop.
make: *** [macosx-standalone-app] Error 2
```



### Steps to reproduce

follow the compile tutorial in the readme

### Details

Provide some additional information:

- USDX version: current master
- Operating System + version: Mac OS X Monterey 12.4

###

brotli is up-to-date I let homebrew reinstall it too

This seems to be related to https://github.com/google/brotli/issues/934 where they recommend using `install_name_tool` to fix it.

--- a/Makefile.osx-helper
+++ b/Makefile.osx-helper
@@ -1,4 +1,5 @@
-getdylibs = $(shell $(OTOOL) -L $(1) | grep version | cut -f 1 -d ' ' | grep -v \/System\/Library | grep -v \/usr\/lib | grep -v executable_path)
+dontcopy = /System/Library/% /usr/lib/% /usr/X11/lib/% /usr/X11R6/lib/% @executable_path/%
+getdylibs = $(filter-out $(dontcopy),$(shell $(OTOOL) -L $(1) | grep version | cut -f 1 -d ' '))
 dest = $(macosx_bundle_path)/MacOS/$(notdir $(1))
 
 # Let's create a list of all libraries used at runtime, starting with the

--- a/Makefile.osx-helper
+++ b/Makefile.osx-helper
@@ -14,6 +14,10 @@ D := $$(firstword $$(foreach rpath,$$(RUNPATH),$$(wildcard $$(patsubst @rpath%,$
 ifeq ($$(D),)
 $$(error dylib $(1) not found in rpath $$(RUNPATH))
 endif
+else
+ifeq ($$(patsubst @loader_path/%,,$$(D)),)
+D := $$(patsubst @loader_path/%,$(2)%,$$(D))
+endif
 endif
 $$(eval RUNPATH_$$(D) := $$$$(RUNPATH))
 NEXTNEWFILES := $(NEXTNEWFILES) $$(filter-out $(FILES) $(NEXTNEWFILES),$$(D))
@@ -21,7 +25,7 @@ endef
 
 define runpathlist
 RUNPATH := $$(RUNPATH_$(1)) $$(filter-out $$(RUNPATH_$(1)),$$(shell $$(OTOOL) -l $(1) | grep -A 2 '\<cmd LC_RPATH\>' | sed -n 's/.* path \(.*\) .offset .*/\1/p'))
-$$(foreach dylib,$$(call getdylibs,$(1)),$$(eval $$(call processload,$$(dylib))))
+$$(foreach dylib,$$(call getdylibs,$(1)),$$(eval $$(call processload,$$(dylib),$$(dir $(1)))))
 endef
 
 define recurse

